### PR TITLE
MET-4566 Unit tests for FullBeanSolrProperties

### PR DESCRIPTION
**Pull request by adolfopeixinho**

### CODE REVIEW

File: metis-indexing/src/test/java/eu/europeana/indexing/solr/property/FullBeanSolrPropertiesTest.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    // proxy setup
+    ProxyImpl proxy = new ProxyImpl();
+    proxy.setAbout("About Proxy");
+    proxy.setEdmType(DocType.TEXT.getEnumNameValue());
+    proxy.setEdmHasType(Map.of("edm types", List.of(DocType.TEXT.getEnumNameValue())));
+    proxy.setEuropeanaProxy(true);
+    proxy.setEdmCurrentLocation(Map.of("Somewhere", List.of("Nowhere")));
+    proxy.setDctermsSpatial(Map.of("Somewhere", List.of("Nowhere")));
+
+    fullBean.setProxies(List.of(proxy));
```
Comment (jeortizquan) [inline metis-indexing/src/test/java/eu/europeana/indexing/solr/property/FullBeanSolrPropertiesTest.java:49]: Add a new test with the following definition of proxies.
```
    List<ProxyImpl> proxies = new ArrayList<>();
    proxies.add(proxy);
    proxies.add(proxy);
    proxies.add(null);
```
to cover this line
eu/europeana/indexing/solr/property/FullBeanSolrProperties.java:48
```
    final String[] types = Optional.ofNullable(fullBean.getProxies()).stream().flatMap(List::stream)
                                   .filter(Objects::nonNull).map(ProxyImpl::getEdmType).filter(Objects::nonNull).distinct()
                                   .toArray(String[]::new);
```

**Review by jeortizquan — CHANGES_REQUESTED** [review]
I see there is a path to cover here.

File: metis-indexing/src/test/java/eu/europeana/indexing/solr/property/FullBeanSolrPropertiesTest.java (hunk 1 above)
Comment (adolfopeixinho) [reply metis-indexing/src/test/java/eu/europeana/indexing/solr/property/FullBeanSolrPropertiesTest.java:48]: This setup throws NPE. Fix is in FullBeanSolrProperties.java:69

File: metis-indexing/src/test/java/eu/europeana/indexing/solr/property/FullBeanSolrPropertiesTest.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    fullBean.setEuropeanaCollectionName(new String[]{"Europeana Collection Name"});
+    fullBean.setTimestampCreated(Date.from(Instant.now().truncatedTo(DAYS)));
+
+    fullBean.setTimestampUpdated(Date.from(Instant.now().truncatedTo(DAYS).plus(1, DAYS)));
+
+    // method to test
+    fullBeanSolrProperties.setProperties(solrInputDocument, fullBean);
+
+    // assertions
+    // List.of("10","33") doesn't work as expected, it adds a whitespace between the two values
```
Comment (adolfopeixinho) [inline metis-indexing/src/test/java/eu/europeana/indexing/solr/property/FullBeanSolrPropertiesTest.java:144]: As stated in the comment: `List.of("10","33")` becomes `["10"," 33"]`

File: metis-indexing/src/test/java/eu/europeana/indexing/solr/property/FullBeanSolrPropertiesTest.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    fullBean.setEuropeanaCollectionName(new String[]{"Europeana Collection Name"});
+    fullBean.setTimestampCreated(Date.from(Instant.now().truncatedTo(DAYS)));
+    fullBean.setTimestampUpdated(Date.from(Instant.now().truncatedTo(DAYS).plus(1, DAYS)));
+
+    // method to test
+    fullBeanSolrProperties.setProperties(solrInputDocument, fullBean);
+
+    // assertions
+    assertEquals(EdmType.TEXT.name(), solrInputDocument.getFieldValue(EdmLabel.PROVIDER_EDM_TYPE.toString()));
+    assertEquals("10,33", solrInputDocument.getFieldValue(EdmLabel.CURRENT_LOCATION_WGS.toString()));
```
Comment (jeortizquan) [inline metis-indexing/src/test/java/eu/europeana/indexing/solr/property/FullBeanSolrPropertiesTest.java:74]: this one is an array.
```
assertEquals(List.of("10,33"), solrInputDocument.getFieldValues(EdmLabel.CURRENT_LOCATION_WGS.toString()));
    assertEquals(List.of("10,33"), solrInputDocument.getFieldValues(EdmLabel.COVERAGE_LOCATION_WGS.toString()));
    assertEquals(List.of("10,33"), solrInputDocument.getFieldValues(EdmLabel.LOCATION_WGS.toString()));
```

File: metis-indexing/src/test/java/eu/europeana/indexing/solr/property/FullBeanSolrPropertiesTest.java [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    place1.setLatitude(10.0f);
+    place1.setLongitude(33.0f);
+    place1.setAltitude(7.0f);
+
+    Place place2 = new PlaceImpl();
+    place2.setAbout("Nowhere");
+    place2.setLatitude(40.0f);
+    place2.setLongitude(53.0f);
+    place2.setAltitude(36.0f);
+
```
[the remark this answers is not in the document]
Comment (jeortizquan) [reply metis-indexing/src/test/java/eu/europeana/indexing/solr/property/FullBeanSolrPropertiesTest.java:109]: it is an array. so you can update the test to this. I think you noticed it before perhaps :smile: ?
```
 assertEquals(List.of("40,53", "10,33"), solrInputDocument.getFieldValues(EdmLabel.CURRENT_LOCATION_WGS.toString()));
    assertEquals(List.of("40,53", "10,33"), solrInputDocument.getFieldValues(EdmLabel.COVERAGE_LOCATION_WGS.toString()));
    assertEquals(List.of("40,53", "10,33"), solrInputDocument.getFieldValues(EdmLabel.LOCATION_WGS.toString()));
```

File: metis-indexing/src/test/java/eu/europeana/indexing/solr/property/FullBeanSolrPropertiesTest.java [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+    fullBean.setEuropeanaCollectionName(new String[]{"Europeana Collection Name"});
+    fullBean.setTimestampCreated(Date.from(Instant.now().truncatedTo(DAYS)));
+    fullBean.setTimestampUpdated(Date.from(Instant.now().truncatedTo(DAYS).plus(1, DAYS)));
+
+    // method to test
+    fullBeanSolrProperties.setProperties(solrInputDocument, fullBean);
+
+    // assertions
+    assertEquals(EdmType.IMAGE.name(), solrInputDocument.getFieldValue(EdmLabel.PROVIDER_EDM_TYPE.toString()));
+    assertEquals("40,53", solrInputDocument.getFieldValue(EdmLabel.CURRENT_LOCATION_WGS.toString()));
```
Comment (jeortizquan) [inline metis-indexing/src/test/java/eu/europeana/indexing/solr/property/FullBeanSolrPropertiesTest.java:147]: see my comment above. ^

**Review by jeortizquan — COMMENTED** [review]
Looks good just a small update.

**Review by jeortizquan — APPROVED** [review]

**Pull request merged**